SwiftCompilerSources/Sources/Optimizer/FunctionPasses/MergeCondFails.swift

```diff
@@ -42,7 +42,8 @@ private func runMergeCondFails(function: Function, context: FunctionPassContext)
         // Do not process arithmetic overflow checks. We typically generate more
         // efficient code with separate jump-on-overflow.
         if !hasOverflowConditionOperand(cfi) &&
-           (condFailToMerge.isEmpty || cfi.message == condFailToMerge.first!.message) {
+           (condFailToMerge.isEmpty ||
+            cfi.location.hasSameSourceLocation(as: condFailToMerge.first!.location)) {
```

Comment:
What about two condfails with different messages on the same source location? That might come up for functions which don't have any debug info, e.g. de-serialized functions.

Reply:
👍